backend/src/main/kotlin/org/darkSolace/muse/userModule/service/UserService.kt

```diff
@@ -0,0 +1,84 @@
+package org.darkSolace.muse.userModule.service
+
+import org.darkSolace.muse.userModule.model.*
+import org.darkSolace.muse.userModule.repository.AvatarRepository
+import org.darkSolace.muse.userModule.repository.UserRepository
+import org.darkSolace.muse.userModule.repository.UserSettingsRepository
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.data.repository.findByIdOrNull
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+
+@Service
+class UserService(
+    @Autowired val userRepository: UserRepository,
+    @Autowired val avatarRepository: AvatarRepository,
+    @Autowired val userSettingsRepository: UserSettingsRepository
+) {
+
+    @Transactional
+    fun createUser(user: User): User {
+        userRepository.save(user)
+        return user
+    }
+
+    @Transactional
+    fun changeAvatar(user: User, avatar: Avatar): User {
+        val changedUser =
+            if (user.id == null) {
+                userRepository.findByUsername(user.username)
+            } else {
+                user
+            }
+
+        if (changedUser.avatar == null) {
+            avatarRepository.save(avatar)
+            changedUser.avatar = avatar
+            userRepository.save(changedUser)
+        } else {
+            changedUser.avatar?.avatarBlob = avatar.avatarBlob
+            userRepository.save(changedUser)
+        }
+
+        return changedUser
+    }
+
+    @Transactional
+    fun getById(id: Long) = userRepository.findByIdOrNull(id)
+
+    @Transactional
+    fun getAll(): List<User> = userRepository.findAll().toList()
+
+    @Transactional
+    fun getAllWithRole(role: Role) = userRepository.findAllByRole(role)
+
+    @Transactional
+    fun deleteUser(user: User) {
+        val deletedUser =
+            if (user.id == null) {
+                userRepository.findByUsername(user.username)
+            } else {
+                user
+            }
+
+        userRepository.delete(deletedUser)
```

Comment:
please add a todo/comment to do something about any possibly existing content they created. 
e.g. attaching their stories to orphan accounts and replacing their username in communication content with 'deleted user' or something